### PR TITLE
Fixed Issue #552

### PR DESCRIPTION
**Pull request by andres-fr**
This PR implements the fix proposed in the issue:

1. Created a small script to convert `Dockerfile` into `Singularity.def` in a way that avoids the `%files` command.
2. Used said script to generate `Singularity.def` (tested on the cluster, works)
3. Updated README docs.

Note that the PR also includes the generated `Singularity.def` file. This refactoring IMO has several advantages for the end user:
* They don't need to install `spython`, one dependency less
* Less steps: also no need to generate the recipe
* The recipe itself is known to work (at least in our cluster)
* If users wish to customize their image creation, they can use the newly provided script as a starting point, much more flexible than the previous command.

Let me know if you have any comments!
Cheers
Andres

### CODE REVIEW

**Comment by andres-fr** [comment]
recheck

**Comment by andres-fr** [comment]
recheck

**Comment by andres-fr** [comment]
I'm not sure I'm doing the CLA right... sorry for the noise, I'll check again tomorrow.

File: docker/scripts/singularity_converter.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+
+import argparse
+#
+import spython
+from spython.main.parse.parsers import get_parser
+from spython.main.parse.writers import get_writer
+
+# globals
+ENTRY_POINT = "/bin/bash"  # seems to be a good default
```
Comment (runame) [inline docker/scripts/singularity_converter.py:20]: Nit: consistently use single quotation marks.

**Review by runame — APPROVED** [review]
LGTM, this should fix the issue. We can just address any remaining problems with follow-up PRs.

**Comment by andres-fr** [comment]
Hi @runame thanks for checking it out.
I've just submitted the signed CLA form to the admins, and pushed the quotation mark fix. Sorry about that one.
Let me know if there is anything else needed from my side to get this PR through.

**Comment by andres-fr** [comment]
Alright, now `isort` should not complain, and resolved the README conflict.
Let me know if any further actions are needed!

File: docker/scripts/singularity_converter.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+"""
+This script is a modification of the
+``spython recipe Dockerfile &> Singularity.def`` command, implemented here:
+github.com/singularityhub/singularity-cli/blob/master/spython/client/recipe.py
+
+It converts the Docker recipy to Singularity, but suppressing any %files
```
Comment (priyakasimbeg) [inline docker/scripts/singularity_converter.py:6]: Nit: recipy -> recipe

**Review by priyakasimbeg — CHANGES_REQUESTED** [review]
Hi Andres, thank you for taking ownership of this issue and helping to get our singularity workflow in better shape!
I just caught one small typo but other than that LGTM.

File: docker/scripts/singularity_converter.py (hunk 2 above)
Comment (andres-fr) [reply docker/scripts/singularity_converter.py:6]: Silly me! thanks for the quick review, I am happy if this helps a bit.

**Review by priyakasimbeg — APPROVED** [review]

**Comment by priyakasimbeg** [comment]
Hi Andres, there is a merge conflict left on this PR after we reorganized the documentation. Can you move your documentation updates to the GETTING_STARTED.md under the section Using Singularity Instead of Docker please? [link to section](https://github.com/mlcommons/algorithmic-efficiency/blob/main/GETTING_STARTED.md#using-singularityapptainer-instead-of-docker)

**Comment by andres-fr** [comment]
Dear @priyakasimbeg, thanks for taking another look into this. I see that you've moved some of the `README` contents into `GETTING_STARTED`. This seemed to include 100% of the merge conflict, so I've:
1. Deleted all merge conflict contents from the README, leaving it exactly as it is on your current `main` branch. No conflicts should remain.
2. Updated  the singularity section from `GETTING_STARTED` with the contents of this PR, plus some minor phrasing updates.

Let me know if this looks OK, and thanks again!

File: README.md [hunk 3]
```diff
@@ -48,14 +48,6 @@ pip3 install -e '.[jax_gpu]' -f 'https://storage.googleapis.com/jax-releases/jax
 pip3 install -e '.[full]'
 ```
 
-*TL;DR to install the PyTorch version for GPU run:*
```
Comment (priyakasimbeg) [inline README.md:51]: @andres-fr  We don't want to remove this command from the README. Could you add it back please?

**Review by priyakasimbeg — CHANGES_REQUESTED** [review]
Thanks for resolving the merge conflicts! I just caught one unintended change in the README.md. I don't think we want to remove the TLDR pytorch installation instructions. Could you add it back please?

File: README.md (hunk 3 above)
Comment (andres-fr) [reply README.md]: I'm so sorry!  You are right, I shouldn't have removed that part. I added it back

**Review by priyakasimbeg — APPROVED** [review]

**Pull request merged**